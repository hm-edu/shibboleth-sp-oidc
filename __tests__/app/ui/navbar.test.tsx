import { afterEach, beforeEach, describe, expect, it, Mock, vi } from 'vitest';
import { cleanup, render, screen } from '@testing-library/react';
import { getServerSession } from 'next-auth';
import Navbar from '@/app/ui/navbar';

vi.mock('next-auth', () => ({
  getServerSession: vi.fn(),
}));

describe('Navbar, when user is not authenticated', () => {
  beforeEach(async () => {
    (getServerSession as Mock).mockReturnValue(null);
    render(await Navbar());
  });

  it('Logo should exists', () => {
    expect(screen.getByLabelText('HM Logo')).toBeDefined();
  });

  it('Logo should match snapshot', () => {
    expect(screen.getByLabelText('HM Logo')).toMatchSnapshot();
  });

  it('Link should exists', () => {
    expect(screen.getByRole('link')).toBeDefined();
  });

  it('Link should redirect to base url', () => {
    expect(screen.getByRole('link').getAttribute('href')).toEqual('/');
  });

  it('Link should match snapshot', () => {
    expect(screen.getByRole('link')).toMatchSnapshot();
  });

  it('Button should exists', () => {
    expect(screen.getByRole('button')).toBeDefined();
  });

  it('Button should match snapshot', () => {
    expect(screen.getByRole('button')).toMatchSnapshot();
  });

  afterEach(() => {
    cleanup();
  });
});

describe('Navbar, when user is authenticated', () => {
  const session = {
    user: {
      id: '1',
      pairwiseId: 'pairwiseId',
    },
    expires: '2021-10-10T10:10:10.000Z',
  };

  beforeEach(async () => {
    (getServerSession as Mock).mockReturnValue(Promise.resolve(session));
    render(await Navbar());
  });

  it('Logo should exists', () => {
    expect(screen.getByLabelText('HM Logo')).toBeDefined();
  });

  it('Logo should match snapshot', () => {
    expect(screen.getByLabelText('HM Logo')).toMatchSnapshot();
  });

  it('Link should exists', () => {
    expect(screen.getByRole('link')).toBeDefined();
  });

  it('Link should redirect to base url', () => {
    expect(screen.getByRole('link').getAttribute('href')).toEqual('/');
  });

  it('Link should match snapshot', () => {
    expect(screen.getByRole('link')).toMatchSnapshot();
  });

  it('Button should exists', () => {
    expect(screen.getByRole('button')).toBeDefined();
  });

  it('Button should match snapshot', () => {
    expect(screen.getByRole('button')).toMatchSnapshot();
  });

  afterEach(() => {
    cleanup();
  });
});
