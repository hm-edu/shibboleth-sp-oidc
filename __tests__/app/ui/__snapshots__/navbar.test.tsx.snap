// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar, when user is authenticated > Button should match snapshot 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit css-e5gali-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  Abmelden
</button>
`;

exports[`Navbar, when user is authenticated > Link should match snapshot 1`] = `
<a
  class="MuiTypography-root MuiTypography-h5 MuiLink-root MuiLink-underlineNone css-q36xxm-MuiTypography-root-MuiLink-root"
  href="/"
>
  Shibboleth OIDC PoC
</a>
`;

exports[`Navbar, when user is authenticated > Logo should match snapshot 1`] = `
<svg
  aria-hidden="true"
  aria-label="HM Logo"
  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-a6ty4p-MuiSvgIcon-root"
  focusable="false"
  viewBox="0 0 138.189 63.78"
  x="0"
  y="0"
>
  <style>
    .st0{fill:#fff}
  </style>
  <path
    class="st0"
    d="m106.299 15.945 15.944-5.315 5.315 15.945-15.944 5.314zM40.942 29.389H15.75v-18.76H10.63v42.52h5.12V33.678h25.192V53.15h5.12V10.63h-5.12zM74.617 46.48H74.5L61.07 10.63h-7.92v42.52h4.883V15.87h.119l14.144 37.28h4.167l14.145-37.28h.12v37.28h4.941V10.63h-7.622z"
  />
</svg>
`;

exports[`Navbar, when user is not authenticated > Button should match snapshot 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit css-e5gali-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  Anmelden
</button>
`;

exports[`Navbar, when user is not authenticated > Link should match snapshot 1`] = `
<a
  class="MuiTypography-root MuiTypography-h5 MuiLink-root MuiLink-underlineNone css-q36xxm-MuiTypography-root-MuiLink-root"
  href="/"
>
  Shibboleth OIDC PoC
</a>
`;

exports[`Navbar, when user is not authenticated > Logo should match snapshot 1`] = `
<svg
  aria-hidden="true"
  aria-label="HM Logo"
  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-a6ty4p-MuiSvgIcon-root"
  focusable="false"
  viewBox="0 0 138.189 63.78"
  x="0"
  y="0"
>
  <style>
    .st0{fill:#fff}
  </style>
  <path
    class="st0"
    d="m106.299 15.945 15.944-5.315 5.315 15.945-15.944 5.314zM40.942 29.389H15.75v-18.76H10.63v42.52h5.12V33.678h25.192V53.15h5.12V10.63h-5.12zM74.617 46.48H74.5L61.07 10.63h-7.92v42.52h4.883V15.87h.119l14.144 37.28h4.167l14.145-37.28h.12v37.28h4.941V10.63h-7.622z"
  />
</svg>
`;
